src/frameworks/uikit/ui_application.rs

```diff
@@ -146,7 +147,20 @@ pub(super) fn UIApplicationMain(
         // The delegate must have been created by this point.
         // While notionally UIApplication does not retain its delegate (see
         // `setDelegate:` above), we do have to retain this first one.
-        let delegate: id = msg![env; ui_application delegate];
+        let delegate: id = {
+            if delegate_class_name == nil {
+                msg![env; ui_application delegate]
+            } else {
+                let delegate_class_name_as_rust_string =
+                    ns_string::to_rust_string(env, delegate_class_name);
+                let delegate_class = env
+                    .objc
+                    .get_known_class(&delegate_class_name_as_rust_string, &mut env.mem);
+                let delegate_object = msg![env; delegate_class new];
+                () = msg![env; ui_application setDelegate:delegate_object];
+                delegate_object
+            }
+        };
```

Comment:
Oh, hmm, when I implemented this I constructed the delegate before loading the nib, but here we do it after.

It's unclear to me whether you can have a named delegate class _and_ a nib file. If you can, I think we might need to construct the delegate first, in case the nib file tries to create connections to this? I'm completely guessing here though.